File: 2239_find_closest_number_to_zero/main.py
from typing import List

class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        close = nums[0]
        for num in nums:
            if abs(num) < abs(close):
                close = num
        for num in nums:
            if abs(close) == abs(num) and close < num:
                return num
        return close

solution = Solution()
print(solution.findClosestNumber(nums=[-4,-2,1,4,8]))

File: 2239_find_closest_number_to_zero/main.go
package main

import (
	"fmt"
	"math"
)

func AbsInt(x int64) int64 {
	if x > 0 {
		return x
	}
	return x * -1
}

func findClosestNumber(nums []int) int {
	close := int64(math.MaxInt64)
	for i := 0; i < len(nums); i++ {
		num := int64(nums[i])
		if AbsInt(num) < AbsInt(close) || (AbsInt(num) == AbsInt(close) && num > close) {
			close = num
		}
	}
	return int(close)
}

func main() {
	nums := []int{2,1,-1}
	res := findClosestNumber(nums)
	fmt.Println(res)
}